core-test/src/test/scala/org/bitcoins/core/crypto/MnemonicCodeTest.scala

```diff
@@ -0,0 +1,951 @@
+package org.bitcoins.core.crypto
+
+import org.bitcoins.core.gen.CryptoGenerators
+import org.bitcoins.core.util.BitcoinSUnitTest
+import org.scalacheck.Prop
+import org.scalatest.Assertion
+import scodec.bits.{BitVector, ByteVector, HexStringSyntax, BinStringSyntax}
+
+import scala.util.Try
+
+/**
+  * [[https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki#test-vectors Trezor]]
+  * test vectors
+  */
+class MnemonicCodeTest extends BitcoinSUnitTest {
+  behavior of "MnemonicCode"
+
+  it must "generate mnemonics of the right length" in {
+    val bits128 = MnemonicCode.getEntropy128Bits
+    MnemonicCode.fromEntropy(bits128).words.length must be(12)
+
+    val bits160 = MnemonicCode.getEntropy160Bits
+    MnemonicCode.fromEntropy(bits160).words.length must be(15)
+
+    val bits192 = MnemonicCode.getEntropy192Bits
+    MnemonicCode.fromEntropy(bits192).words.length must be(18)
+
+    val bits224 = MnemonicCode.getEntropy224Bits
+    MnemonicCode.fromEntropy(bits224).words.length must be(21)
+
+    val bits256 = MnemonicCode.getEntropy256Bits
+    MnemonicCode.fromEntropy(bits256).words.length must be(24)
+  }
+
+  it must "fail to generate mnemonics from entropy of bad length" in {
+    def testEntropyFailure(entropy: BitVector): Assertion = {
+      val short = entropy.drop(1)
+      val long = entropy ++ bin"1"
+      Try(MnemonicCode.fromEntropy(short)).isFailure must be(true)
```

Comment:
I think using `assertThrows[ExceptionTypeHere]` is nice for things like this because then it will fail the test if you don't get the expected failure